core/src/validator.rs

```diff
@@ -607,9 +630,39 @@ fn new_banks_from_blockstore(
     )
     .unwrap_or_else(|err| {
         error!("Failed to load ledger: {:?}", err);
-        std::process::exit(1);
+        process::exit(1);
     });
 
+    let tower = match restored_tower {
+        Ok(mut tower) => {
+            // The tower root can be older if the validator booted from a newer snapshot, so
+            // tower lockouts may need adjustment
+            tower.adjust_lockouts_if_newer_root(bank_forks.root());
+            tower
+        }
+        Err(err) => {
+            if config.require_tower
+                && empty_vote_account(&bank_forks.working_bank(), &vote_account) != Some(true)
+            {
+                error!("Tower restore failed: {:?}", err);
+                process::exit(1);
+            }
+            info!("Rebuilding tower from the latest vote account");
```

Comment:
@t-nelson, do you remember what the original design was for the case where tower restoration failed? The danger with restoring from the latest vote account is that it could be missing some votes that have been submitted onto the network, but have not landed in any bank.

I think we said we should either signal to the user to make a new vote account and eat the warmup/cooldown for staking, or have them accept the risk of potentially submitting conflicting votes.

Reply:
Ultimately I think it comes down to what the slashing design is (which we don't have finalized, so it makes i hard to reason about this). For instance, if we decide slashing can only occur for votes that land in a bank, and that all votes before the root are not slashable (not sure if this is a reasonable assumption) and you have a reasonable guess for the range of your last vote, maybe you can just wait for a root to be made that is sufficiently far past your estimated last vote... just food for thought

Reply:
Hmm... Yeah, there should definitely be user intervention here (at least by the time slashing is enabled.  It _is_ behind a CLI flag.  Maybe it'd be sufficient to reverse its default behavior? That is require the tower by default.

Reply:
agreed, there should be some level of intervention notification here